packages/storybook/examples/core/Modal.stories.js

```diff
@@ -0,0 +1,160 @@
+
+import Button from '@ichef/gypcrete/src/Button';
+import HeaderRow from '@ichef/gypcrete/src/HeaderRow';
+import TextLabel from '@ichef/gypcrete/src/TextLabel';
+
+import ContainsColumnView from './SplitView/ContainsColumnView';
+
+export default {
+    title: '@ichef/gypcrete|Modal',
+    component: PureModal,
+    subcomponents: {
+        'renderToLayer()': Modal
+    },
+};
+
+export function ClosableModal({
+    children = 'Modal Content',
+    title = 'Closable Modal',
+    ...modalProps
+}) {
+    const [modalOpen, setModalOpen] = useState(false);
+
+    const handleModalOpen = () => {
+        setModalOpen(true);
+    };
+
+    const handleModalClose = () => {
+        action('cancel')();
+        setModalOpen(false);
+    };
+
+    // For storybook demo usage we have to move <ModalHeader> inside <ClosableModal>,
+    // or storybook won't show the full jsx code.
+    // However, in real use case you may want to extract this out.
+
+    const ModalHeader = ({ onCancel, labelText }) => {
+        const cancelBtn = <Button color="black" basic="Cancel" onClick={onCancel} />;
+        const filterBtn = <Button bold basic="Filter" onClick={action('filter')} />;
+        const label = <TextLabel align="center" basic={labelText} />;
+
+        return (
+            <HeaderRow
+                left={cancelBtn}
+                center={label}
+                right={filterBtn} />
+        );
+    };
+
+    ModalHeader.propTypes = {
+        labelText: PropTypes.string,
+        onCancel: PropTypes.func,
+    };
+
+    ModalHeader.defaultProps = {
+        labelText: '',
+        onCancel: () => {},
+    };
+
+    const header = (
+        <ModalHeader
+            onCancel={handleModalClose}
+            labelText={title}
+        />
+    );
+    if (!modalOpen) {
+        return (
+            <div>
+                <Button
+                    solid
+                    color="blue"
+                    onClick={handleModalOpen}
+                    style={{ display: 'inline-block' }}
+                >
+                    Open Modal
+                </Button>
+            </div>
+        );
+    }
+
+    return (
+        <Modal
+            header={header}
+            onClose={handleModalClose}
+            {...modalProps}
+        >
+            {children}
+        </Modal>
+    );
+}
+
+ClosableModal.propTypes = {
+    title: PropTypes.string,
+};
+
+
+export function SplitViewModal() {
+    return (
+        /* See the Closable Modal example */
+        <ClosableModal title="With <SplitView>" flexBody bodyPadding={{ bottom: 0 }}>
+            {/* See <SplitView> => `contains <ColumnView>` doc for <ContainsColumnView> */}
+            <ContainsColumnView />
+        </ClosableModal>
+    );
+}
+
+SplitViewModal.story = {
+    name: 'With <SplitView>',
+};
+
+export function CenteredModal() {
+    return (
+        /* See the Closable Modal example */
+        <ClosableModal title="Vertically-centered modal" centered>
```

Comment:
There seems a bug -- `centered` Modal has wider width because we use `display: flex` to center it so the `width` property is ignored. Will fix this in another PR.